### PR TITLE
Support axis-specific location locks

### PR DESCRIPTION
**Pull request by sehilyi**
## Description

This PR adds a way to link specific axes between views in `viewConfig`. For example, one can do cross-axis lock, i.e., the vertical axis on a view ⇔ the horizontal axis on another, or lock only the x-axes between two views. 

Config-wise, this is enabled by additionally accepting "object" properties for `locationLocks.lockByViewUid.uid`, in addition to "string":

### Original ViewConfig
```js
// all views are locked
"locationLocks": {
    "locksByViewUid": {
      "view-1": "lock-1",  
      "view-3": "lock-1",
    }
  },
```
### Newly Accepted ViewConfig
```js
// the x-axis of "view-1" is linked with the y-axis of "view-2"
"locationLocks": {
    "locksByViewUid": {
      "view-1": { "x": { "lock": "lock-1", "axis": "y" } },  
      "view-2": { "y": { "lock": "lock-1", "axis": "x" } }, 
    }
  },
```

Since this feature is used only when user specify "objects", I think this updates will not affect any to existing viewconfigs that users have.

This can be useful for some usage scenarios that require more complex coordinate multiple view settings (e.g., linking between a vertical-track-only view and a horizontal-track-only view). In Gosling, we would like to implement vertical band connection like in [GIVE](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1465-6) by using this functionality.

## Example

A simple example with three views (Blue, Green, Red). The x-axes of Blue and Red are locked, and the x-axis of Green is locked to the y-axis of Red.

Aug-13-2021 16-41-38

```js
"locationLocks": {
    "locksByViewUid": {
      "view-1": { "x": { "lock": "lock-1", "axis": "x" } },  
      // the x-axis of "view-2" is locked to the y-axis of "lock-2"
      "view-2": { "x": { "lock": "lock-2", "axis": "y" } }, 
      "view-3": { "y": { "lock": "lock-2", "axis": "x" }, "x": { "lock": "lock-1", "axis": "x" }, },
    },
    "locksDict": {
      "lock-1": {
        "view-1": [...],
        "view-3": [...],
        "uid": "lock-1"
      },
      "lock-2": {
        "view-2": [...],
        "view-3": [...],
        "uid": "lock-2"
      }
    }
  },
```

<details>
<summary>Full View Config</summary>

```js
{
  "zoomFixed": false,
  "views": [
    {
      "layout": {
        "w": 6,
        "h": 12,
        "x": 0,
        "y": 0,
        "moved": false,
        "static": false
      },
      "uid": "view-1",
      "initialYDomain": [
        1500488288.4451075,
        1501904775.6260266
      ],
      "autocompleteSource": "/api/v1/suggest/?d=OHJakQICQD6gTD7skx4EWA&",
      "initialXDomain": [
        1500275348.1028004,
        1501641545.7980063
      ],
      "tracks": {
        "bottom": [],
        "top": [
          {
            "uid": "OHJakQICQD6gTD7skx4EWA",
            "tilesetUid": "OHJakQICQD6gTD7skx4EWA",
            "server": "//higlass.io/api/v1",
            "height": 48,
            "type": "horizontal-gene-annotations",
            "options": {
              "labelColor": "black",
              "plusStrandColor": "blue",
              "name": "Gene Annotations (hg19)",
              "labelPosition": "hidden",
              "minusStrandColor": "red",
              "fontSize": 10,
              "labelBackgroundColor": "#ffffff",
              "labelLeftMargin": 0,
              "labelRightMargin": 0,
              "labelTopMargin": 0,
              "labelBottomMargin": 0,
              "minHeight": 24,
              "trackBorderWidth": 0,
              "trackBorderColor": "black",
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "geneAnnotationHeight": 16,
              "geneLabelPosition": "outside",
              "geneStrandSpacing": 4
            },
            "width": 321
          },
          {
            "uid": "X4e_1DKiQHmyghDa6lLMVA",
            "height": 30,
            "chromInfoPath": "//s3.amazonaws.com/pkerp/data/hg19/chromSizes.tsv",
            "type": "horizontal-chromosome-labels",
            "options": {
              "color": "#808080",
              "stroke": "#ffffff",
              "fontSize": 12,
              "fontIsLeftAligned": false,
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "reverseOrientation": false
            }
          }
        ],
        "right": [],
        "center": [
          {
            "uid": "c1",
            "height": 200,
            "type": "combined",
            "options": {},
            "contents": [
              {
                "uid": "GjuZed1ySGW1IzZZqFB9BA",
                "tilesetUid": "CQMd6V_cRw6iCI_-Unl3PQ",
                "server": "//higlass.io/api/v1",
                "type": "heatmap",
                "options": {
                  "maxZoom": null,
                  "name": "Rao et al. (2014) GM12878 MboI (allreps) 1kb",
                  "colorRange": [
                    "rgba(255,255,255,1)",
                    "rgba(0,0,0,1)"
                  ],
                  "labelPosition": "bottomRight",
                  "colorbarLabelsPosition": "inside",
                  "colorbarPosition": "topRight",
                  "backgroundColor": "#eeeeee",
                  "labelLeftMargin": 0,
                  "labelRightMargin": 0,
                  "labelTopMargin": 0,
                  "labelBottomMargin": 0,
                  "labelShowResolution": true,
                  "labelShowAssembly": true,
                  "colorbarBackgroundColor": "#ffffff",
                  "minWidth": 100,
                  "minHeight": 100,
                  "trackBorderWidth": 0,
                  "trackBorderColor": "black",
                  "heatmapValueScaling": "log",
                  "showMousePosition": false,
                  "mousePositionColor": "#000000",
                  "showTooltip": false,
                  "extent": "full",
                  "zeroValueColor": null,
                  "scaleStartPercent": "0.00000",
                  "scaleEndPercent": "1.00000"
                },
                "transforms": [
                  {
                    "name": "ICE",
                    "value": "weight"
                  }
                ]
              }
            ]
          }
        ],
        "left": [
          {
            "tilesetUid": "OHJakQICQD6gTD7skx4EWA",
            "server": "//higlass.io/api/v1",
            "width": 45,
            "type": "vertical-gene-annotations",
            "options": {
              "labelColor": "black",
              "plusStrandColor": "blue",
              "name": "Gene Annotations (hg19)",
              "labelPosition": "bottomRight",
              "minusStrandColor": "red",
              "fontSize": 10,
              "labelBackgroundColor": "#ffffff",
              "labelLeftMargin": 0,
              "labelRightMargin": 0,
              "labelTopMargin": 0,
              "labelBottomMargin": 0,
              "minHeight": 24,
              "trackBorderWidth": 0,
              "trackBorderColor": "black",
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "geneAnnotationHeight": 16,
              "geneLabelPosition": "outside",
              "geneStrandSpacing": 4
            },
            "uid": "dqBTMH78Rn6DeSyDBoAEXw",
            "height": 284
          },
          {
            "uid": "RHdQK4IRQ7yJeDmKWb7Pcg",
            "width": 30,
            "chromInfoPath": "//s3.amazonaws.com/pkerp/data/hg19/chromSizes.tsv",
            "type": "vertical-chromosome-labels",
            "options": {
              "color": "#808080",
              "stroke": "#ffffff",
              "fontSize": 12,
              "fontIsLeftAligned": false,
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "reverseOrientation": false
            }
          }
        ],
        "whole": [],
        "gallery": []
      },
      "chromInfoPath": "//s3.amazonaws.com/pkerp/data/hg19/chromSizes.tsv",
      "genomePositionSearchBox": {
        "autocompleteServer": "//higlass.io/api/v1",
        "autocompleteId": "OHJakQICQD6gTD7skx4EWA",
        "chromInfoServer": "//higlass.io/api/v1",
        "chromInfoId": "hg19",
        "visible": true
      }
    },
    {
      "layout": {
        "w": 6,
        "h": 12,
        "x": 6,
        "y": 0,
        "moved": false,
        "static": false
      },
      "uid": "view-2",
      "initialYDomain": [
        1500486193.0498662,
        1501906871.0212667
      ],
      "autocompleteSource": "/api/v1/suggest/?d=OHJakQICQD6gTD7skx4EWA&",
      "initialXDomain": [
        1500281634.288512,
        1501635259.612295
      ],
      "tracks": {
        "bottom": [],
        "top": [
          {
            "uid": "OHJakQICQD6gTD7skx4EWA",
            "tilesetUid": "OHJakQICQD6gTD7skx4EWA",
            "server": "//higlass.io/api/v1",
            "height": 47,
            "type": "horizontal-gene-annotations",
            "options": {
              "labelColor": "black",
              "plusStrandColor": "blue",
              "name": "Gene Annotations (hg19)",
              "labelPosition": "hidden",
              "minusStrandColor": "red",
              "fontSize": 10,
              "labelBackgroundColor": "#ffffff",
              "labelLeftMargin": 0,
              "labelRightMargin": 0,
              "labelTopMargin": 0,
              "labelBottomMargin": 0,
              "minHeight": 24,
              "trackBorderWidth": 0,
              "trackBorderColor": "black",
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "geneAnnotationHeight": 16,
              "geneLabelPosition": "outside",
              "geneStrandSpacing": 4
            },
            "width": 321
          },
          {
            "uid": "X4e_1DKiQHmyghDa6lLMVA",
            "height": 30,
            "chromInfoPath": "//s3.amazonaws.com/pkerp/data/hg19/chromSizes.tsv",
            "type": "horizontal-chromosome-labels",
            "options": {
              "color": "#808080",
              "stroke": "#ffffff",
              "fontSize": 12,
              "fontIsLeftAligned": false,
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "reverseOrientation": false
            }
          }
        ],
        "right": [],
        "center": [
          {
            "type": "combined",
            "uid": "QN8h5oodTfOI-ZLTfd_0Bg",
            "contents": [
              {
                "uid": "d4bx9LAFRwWAcVmVqVf_1A",
                "tilesetUid": "QvdMEvccQuOxKTEjrVL3wA",
                "server": "//higlass.io/api/v1",
                "width": 20,
                "height": 20,
                "type": "heatmap",
                "options": {
                  "maxZoom": null,
                  "labelPosition": "bottomRight",
                  "colorRange": [
                    "rgba(0,255,0,1)",
                    "rgba(0,0,0,1)"
                  ],
                  "name": "Rao et al. (2014) K562 MboI (allreps) 1kb",
                  "colorbarLabelsPosition": "inside",
                  "colorbarPosition": "topRight",
                  "backgroundColor": "#eeeeee",
                  "labelLeftMargin": 0,
                  "labelRightMargin": 0,
                  "labelTopMargin": 0,
                  "labelBottomMargin": 0,
                  "labelShowResolution": true,
                  "labelShowAssembly": true,
                  "colorbarBackgroundColor": "#ffffff",
                  "minWidth": 100,
                  "minHeight": 100,
                  "trackBorderWidth": 0,
                  "trackBorderColor": "black",
                  "heatmapValueScaling": "log",
                  "showMousePosition": false,
                  "mousePositionColor": "#000000",
                  "showTooltip": false,
                  "extent": "full",
                  "zeroValueColor": null,
                  "scaleStartPercent": "0.00000",
                  "scaleEndPercent": "1.00000"
                },
                "transforms": [
                  {
                    "name": "ICE",
                    "value": "weight"
                  }
                ]
              }
            ],
            "options": {}
          }
        ],
        "left": [
          {
            "tilesetUid": "OHJakQICQD6gTD7skx4EWA",
            "server": "//higlass.io/api/v1",
            "width": 48,
            "type": "vertical-gene-annotations",
            "options": {
              "labelColor": "black",
              "plusStrandColor": "blue",
              "name": "Gene Annotations (hg19)",
              "labelPosition": "bottomRight",
              "minusStrandColor": "red",
              "fontSize": 10,
              "labelBackgroundColor": "#ffffff",
              "labelLeftMargin": 0,
              "labelRightMargin": 0,
              "labelTopMargin": 0,
              "labelBottomMargin": 0,
              "minHeight": 24,
              "trackBorderWidth": 0,
              "trackBorderColor": "black",
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "geneAnnotationHeight": 16,
              "geneLabelPosition": "outside",
              "geneStrandSpacing": 4
            },
            "uid": "dqBTMH78Rn6DeSyDBoAEXw",
            "height": 285
          },
          {
            "uid": "RHdQK4IRQ7yJeDmKWb7Pcg",
            "width": 30,
            "chromInfoPath": "//s3.amazonaws.com/pkerp/data/hg19/chromSizes.tsv",
            "type": "vertical-chromosome-labels",
            "options": {
              "color": "#808080",
              "stroke": "#ffffff",
              "fontSize": 12,
              "fontIsLeftAligned": false,
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "reverseOrientation": false
            }
          }
        ],
        "whole": [],
        "gallery": []
      },
      "chromInfoPath": "//s3.amazonaws.com/pkerp/data/hg19/chromSizes.tsv",
      "genomePositionSearchBox": {
        "autocompleteServer": "//higlass.io/api/v1",
        "autocompleteId": "OHJakQICQD6gTD7skx4EWA",
        "chromInfoServer": "//higlass.io/api/v1",
        "chromInfoId": "hg19",
        "visible": true
      }
    },
    {
      "layout": {
        "w": 12,
        "h": 12,
        "x": 0,
        "y": 6,
        "moved": false,
        "static": false
      },
      "uid": "view-3",
      "initialYDomain": [
        1500486193.0498662,
        1501906871.0212667
      ],
      "autocompleteSource": "/api/v1/suggest/?d=OHJakQICQD6gTD7skx4EWA&",
      "initialXDomain": [
        1500281634.288512,
        1501635259.612295
      ],
      "tracks": {
        "bottom": [],
        "top": [
          {
            "tilesetUid": "OHJakQICQD6gTD7skx4EWA",
            "server": "//higlass.io/api/v1",
            "height": 47,
            "type": "horizontal-gene-annotations",
            "options": {
              "labelColor": "black",
              "plusStrandColor": "blue",
              "name": "Gene Annotations (hg19)",
              "labelPosition": "hidden",
              "minusStrandColor": "red",
              "fontSize": 10,
              "labelBackgroundColor": "#ffffff",
              "labelLeftMargin": 0,
              "labelRightMargin": 0,
              "labelTopMargin": 0,
              "labelBottomMargin": 0,
              "minHeight": 24,
              "trackBorderWidth": 0,
              "trackBorderColor": "black",
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "geneAnnotationHeight": 16,
              "geneLabelPosition": "outside",
              "geneStrandSpacing": 4
            },
            "width": 321
          },
          {
            "height": 30,
            "chromInfoPath": "//s3.amazonaws.com/pkerp/data/hg19/chromSizes.tsv",
            "type": "horizontal-chromosome-labels",
            "options": {
              "color": "#808080",
              "stroke": "#ffffff",
              "fontSize": 12,
              "fontIsLeftAligned": false,
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "reverseOrientation": false
            }
          }
        ],
        "right": [],
        "center": [
          {
            "type": "combined",
            "contents": [
              {
                "tilesetUid": "QvdMEvccQuOxKTEjrVL3wA",
                "server": "//higlass.io/api/v1",
                "width": 20,
                "height": 20,
                "type": "heatmap",
                "options": {
                  "maxZoom": null,
                  "labelPosition": "bottomRight",
                  "colorRange": [
                    "rgba(255,0,0,1)",
                    "rgba(0,0,0,1)"
                  ],
                  "name": "Rao et al. (2014) K562 MboI (allreps) 1kb",
                  "colorbarLabelsPosition": "inside",
                  "colorbarPosition": "topRight",
                  "backgroundColor": "#eeeeee",
                  "labelLeftMargin": 0,
                  "labelRightMargin": 0,
                  "labelTopMargin": 0,
                  "labelBottomMargin": 0,
                  "labelShowResolution": true,
                  "labelShowAssembly": true,
                  "colorbarBackgroundColor": "#ffffff",
                  "minWidth": 100,
                  "minHeight": 100,
                  "trackBorderWidth": 0,
                  "trackBorderColor": "black",
                  "heatmapValueScaling": "log",
                  "showMousePosition": false,
                  "mousePositionColor": "#000000",
                  "showTooltip": false,
                  "extent": "full",
                  "zeroValueColor": null,
                  "scaleStartPercent": "0.00000",
                  "scaleEndPercent": "1.00000"
                },
                "transforms": [
                  {
                    "name": "ICE",
                    "value": "weight"
                  }
                ]
              }
            ],
            "options": {}
          }
        ],
        "left": [
          {
            "tilesetUid": "OHJakQICQD6gTD7skx4EWA",
            "server": "//higlass.io/api/v1",
            "width": 48,
            "type": "vertical-gene-annotations",
            "options": {
              "labelColor": "black",
              "plusStrandColor": "blue",
              "name": "Gene Annotations (hg19)",
              "labelPosition": "bottomRight",
              "minusStrandColor": "red",
              "fontSize": 10,
              "labelBackgroundColor": "#ffffff",
              "labelLeftMargin": 0,
              "labelRightMargin": 0,
              "labelTopMargin": 0,
              "labelBottomMargin": 0,
              "minHeight": 24,
              "trackBorderWidth": 0,
              "trackBorderColor": "black",
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "geneAnnotationHeight": 16,
              "geneLabelPosition": "outside",
              "geneStrandSpacing": 4
            },
            "height": 285
          },
          {
            "width": 30,
            "chromInfoPath": "//s3.amazonaws.com/pkerp/data/hg19/chromSizes.tsv",
            "type": "vertical-chromosome-labels",
            "options": {
              "color": "#808080",
              "stroke": "#ffffff",
              "fontSize": 12,
              "fontIsLeftAligned": false,
              "showMousePosition": false,
              "mousePositionColor": "#000000",
              "reverseOrientation": false
            }
          }
        ],
        "whole": [],
        "gallery": []
      },
      "chromInfoPath": "//s3.amazonaws.com/pkerp/data/hg19/chromSizes.tsv",
      "genomePositionSearchBox": {
        "autocompleteServer": "//higlass.io/api/v1",
        "autocompleteId": "OHJakQICQD6gTD7skx4EWA",
        "chromInfoServer": "//higlass.io/api/v1",
        "chromInfoId": "hg19",
        "visible": true
      }
    }
  ],
  "editable": true,
  "exportViewUrl": "/api/v1/viewconfs",
  "zoomLocks": {
    "locksByViewUid": {
      "view-1": "lock-1",
      "view-2": "lock-1",
      "view-3": "lock-1"
    },
    "locksDict": {
      "lock-1": {
        "view-1": [
          1550000000,
          1550000000,
          3380588.876772046
        ],
        "view-2": [
          1550000000.0000002,
          1549999999.9999993,
          3380588.876772046
        ],
        "view-3": [
          1550000000.0000002,
          1549999999.9999993,
          3380588.876772046
        ],
        "uid": "lock-1"
      }
    }
  },
  "trackSourceServers": [
    "//higlass.io/api/v1"
  ],
  "locationLocks": {
    "locksByViewUid": {
      "view-1": { "x": { "lock": "lock-1", "axis": "x" } },
      "view-2": { "x": { "lock": "lock-2", "axis": "y" } },
      "view-3": { "y": { "lock": "lock-2", "axis": "x" }, "x": { "lock": "lock-1", "axis": "x" }}
    },
    "locksDict": {
      "lock-1": {
        "view-1": [
          1550000000.0000002,
          1549999999.9999993,
          3380588.876772046
        ],
        "view-3": [
          1550000000.0000002,
          1549999999.9999993,
          3380588.876772046
        ],
        "uid": "lock-1"
      },
      "lock-2": {
        "view-2": [
          1550000000.0000002,
          1549999999.9999993,
          3380588.876772046
        ],
        "view-3": [
          1550000000.0000002,
          1549999999.9999993,
          3380588.876772046
        ],
        "uid": "lock-2"
      }
    }
  },
  "valueScaleLocks": {
    "locksByViewUid": {},
    "locksDict": {}
  }
}

```
</details>

## Checklist

- [ ] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [x] Unit tests added or updated
- [x] Documentation added or updated
- [x] Example(s) added or updated
- [x] Update schema.json if there are changes to the viewconf JSON structure format
- [x] Screenshot for visual changes (e.g. new tracks or UI changes)
- [x] Updated CHANGELOG.md

### CODE REVIEW

**Review by pkerpedjiev — COMMENTED** [review]
This looks like it's super useful! Thank you for putting it together. A few thoughts:

1. Would you mind adding docs in the viewconfig section explaining the new definition format?
2. It doesn't seem like you've updated the viewconf schema: https://github.com/higlass/higlass/blob/develop/app/schema.json
3. Would you mind adding some tests?
4. Is it possible to create deadlocks or contradictions using the new functionality? I.e. somehow create a circular chain of locks.
5. Will there be a user interface for setting these locks or is it meant to be only available in the viewconf? Either is fine, just want to make sure.

**Comment by sehilyi** [comment]
@pkerpedjiev, thank you so much for the review and the comments!

I will update the code reflecting on the point 1-3. 

Regarding 3: That's a good question. I think this schema is safe from creating the circular chain of locks. To make the chain cyclic, one should be able to link a single axis of a view to multiple axes with different location locks (e.g., Axis A <-> Axis B and Axis A <-> Axis C). But, linking multiple locks to a single axis is not possible by the current schema:

```js
"view-1": {
   "x": { "lock": "lock-1", ... }, 
   "x": { "lock": "lock-2", ...} // Error, "x" cannot be defined multiple times
}
```

Regarding 4: I think UI-based configuration will be useful for end-users since it might be challenging to configure the location locks accurately by themselves. But, I think supporting intuitive UI requires more careful design considering the increased complexity of the location locks. Since our primary use case in Gosling is to use the `viewConfig`, I can think further about the UI support and open a separate GitHub issue to discuss it. Does this sound good to you?

File: docs/view_config.rst [hunk 1]
```diff
@@ -285,6 +285,94 @@ database, assigns them a uid, and makes them accessible through its API at
         "exportViewUrl": "/api/v1/viewconfs",
     }
 
+zoomLocks
+^^^^^^^^^
+
+In HiGlass, the zoom and location of multiple views can be synchronized for the
```
Comment (pkerpedjiev) [inline docs/view_config.rst:291]:
```suggestion
In HiGlass, the zoom and location of multiple views can be synchronized for 
```

File: app/scripts/HiGlassComponent.js [hunk 2]
```diff
@@ -1724,6 +1727,136 @@ class HiGlassComponent extends React.Component {
...
+
+        if (key === uid) {
+          // no need to notify oneself that the scales have changed
+          continue;
+        }
+
+        // const dx = value[0] - lockGroup[uid][0];
+        const dy = value[1] - lockGroup[uid][1];
+
+        // const newCenterX = centerX + dx;
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassComponent.js:1831]: delete if not used

File: app/scripts/HiGlassComponent.js [hunk 3]
```diff
@@ -1724,6 +1727,136 @@ class HiGlassComponent extends React.Component {
...
+          this.xScales[key],
+          this.yScales[key],
+        );
+
+        if (key === uid) {
+          // no need to notify oneself that the scales have changed
+          continue;
+        }
+
+        // const dx = value[0] - lockGroup[uid][0];
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassComponent.js:1828]: delete if not used

File: app/scripts/HiGlassComponent.js [hunk 4]
```diff
@@ -1724,6 +1727,136 @@ class HiGlassComponent extends React.Component {
...
+        if (key === uid) {
+          // no need to notify oneself that the scales have changed
+          continue;
+        }
+
+        const dx = value[0] - lockGroup[uid][0];
+        // const dy = value[1] - lockGroup[uid][1];
+
+        const newCenterX = centerX + dx;
+        // const newCenterY = centerY + dy;
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassComponent.js:1767]: delete if not used

File: app/scripts/HiGlassComponent.js [hunk 5]
```diff
@@ -1724,6 +1727,136 @@ class HiGlassComponent extends React.Component {
...
+          this.yScales[key],
+        );
+
+        if (key === uid) {
+          // no need to notify oneself that the scales have changed
+          continue;
+        }
+
+        const dx = value[0] - lockGroup[uid][0];
+        // const dy = value[1] - lockGroup[uid][1];
```
Comment (pkerpedjiev) [inline app/scripts/HiGlassComponent.js:1764]: delete if not used

**Review by pkerpedjiev — APPROVED** [review]
A few minor comments about deleting some comments but otherwise looks good 👍 Thanks for adding!

**Pull request merged**